### ISSUE TITLE
Plugin is not working when location service is off

### ISSUE THREAD

**Issue by UrviNivida** [rekab-app/background_locator#128]
when i switch off the location service from top menu in mobile  , then this plugin is not working neither giving me popup to on the location permission, i want to fetch the location or gps status and due to above issue i m not able to

**Comment by anirudhsharma392** [comment]
Actually I am facing the same issue,
I think we need this one [here](https://github.com/rekab-app/background_locator/wiki/Use-other-plugins-in-callback)
and we can get the status with the help of this package [here](https://pub.dev/packages/geolocator)

Now the problem is don't know how to add the above package to get the GPS status 
my current code is

import io.flutter.app.FlutterApplication
import io.flutter.plugin.common.PluginRegistry
import io.flutter.plugins.pathprovider.PathProviderPlugin
import io.flutter.view.FlutterMain
import rekab.app.background_locator.LocatorService
//don't know how to import this [package](https://pub.dev/packages/geolocator)


class Application : FlutterApplication(), PluginRegistry.PluginRegistrantCallback {
    override fun onCreate() {
        super.onCreate()
        LocatorService.setPluginRegistrant(this)
        FlutterMain.startInitialization(this)
    }

    override fun registerWith(registry: PluginRegistry?) {
        if (!registry!!.hasPlugin("io.flutter.plugins.pathprovider")) {
            PathProviderPlugin.registerWith(registry!!.registrarFor("io.flutter.plugins.pathprovider"))
        }

    }
}

Any help would be appreciated,
Thanks

**Comment by UrviNivida** [comment]
> Actually I was facing the same issue and cannot get the status of GPS,
> I think you need this one [here](https://github.com/rekab-app/background_locator/wiki/Use-other-plugins-in-callback)
> and we can get the status with the help of this package [here](https://pub.dev/packages/location)
> first you need to add the required plugin using like this
> 
> ```
> package jsimple.hr_on_tips
> import io.flutter.app.FlutterApplication
> import io.flutter.plugin.common.PluginRegistry
> import io.flutter.view.FlutterMain
> import rekab.app.background_locator.LocatorService
> import com.lyokone.location.LocationPlugin
> 
> class Application : FlutterApplication(), PluginRegistry.PluginRegistrantCallback {
>     override fun onCreate() {
>         super.onCreate()
>         LocatorService.setPluginRegistrant(this)
>         FlutterMain.startInitialization(this)
>     }
> 
>     override fun registerWith(registry: PluginRegistry?) {
>        
>         if (!registry!!.hasPlugin("com.lyokone.location.LocationPlugin")) {
>             LocationPlugin.registerWith(registry!!.registrarFor("com.lyokone.location.LocationPlugin"))
>         }
>     }
> 
> }
> ```
> 
> and then simply check GPS status in callback
> 
> ```
>  var location = loc.Location();
>       if (!await location.serviceEnabled()) {
> ///do something here
>       }else{
> ///service not available
> }
> ```

**Comment by UrviNivida** [comment]
yes i got your point but the isssue is that my web service is calling in "LocationServiceRepository" class so , when i stop location service from top mpbile menu then the callback method ' Future<void> callback(LocationDto locationDto)'  stop so i wont b able to send data in my web service that gps is off.

**Comment by anirudhsharma392** [comment]
Ohh, we need to call the api when gps is off, but callback function never calls itself if GPS is off. THAT DEADLOCK !!!!!!!!!!!!
Did you find out any workaround to it.
